comms/src/parse/Requests.ts

```diff
@@ -0,0 +1,123 @@
+import { Instruction } from "../debug/Instructions";
+
+export interface StackValue {
+    idx: number,
+    type: string;
+    value: number | bigint;
+}
+
+export interface CallbackMapping {
+    callbackid: string;
+    tableIndexes: number[]
+}
+
+export interface InterruptEvent {
+    topic: string;
+    payload: string;
+}
+
+export interface Frame {
+    type: number;
+    fidx: string;
+    sp: number;
+    fp: number;
+    block_key: number;
+    ra: number;
+    idx: number;
+}
+
+export interface Table {
+    max: number;
+    init: number;
+    elements: number[];
+}
+
+export interface Memory {
+    pages: number;
+    max: number;
+    init: number;
+    bytes: Uint8Array;
+}
+
+export interface BRTable {
```

Comment:
It is not the same. A Wasm table is one in which you store the function identifiers whereas a BRTable is something specific to WARDuino that is used for branching operations such as `loop`, `br_if` etc.

I need the BRTable to make out-of-place possible. See comment here above

Reply:
But the TypeScript interface literally has the same fields.

> whereas a BRTable is something specific to WARDuino that is used for branching operations such as loop, br_if etc.

Are you sure? That seems like a wasm thing. Both loop and br_if are wasm instructions.

Tables in wasm are vectors of opaque values, they do not have to hold function identifiers.
You can have other tables as well.

Reply:
There seems to be confusion about what we are talking about. To clarify in this code we have two interfaces that share the word table in the interface name:
- Table: the opaque vectors that you mentioned and that are indeed part of the Wasm spec.
- BRTable: something that I believe is WARDuinospecific and that is introduced in WARDuino to support the Wasm instructions br_if, loop, etc. The BRTable corresponds with this struct field from the Module struct https://github.com/TOPLLab/WARDuino/blob/c586b8ea7ee808011ff6aceec6b3441896b4e41a/src/WARDuino.h#L170

Reply:
ok but apparently you can't remove it because the field is called labels in the JSON, right?
So keep it. But we should make a WARDuino namespace alongside the WASM one, for the Branch table, Callback mapping, etc.